CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -72,8 +72,16 @@ function Patient:onClick(ui, button)
     else
       local hospital = self.hospital or self.world:getLocalPlayerHospital()
       local epidemic = hospital and hospital.epidemic
-      if epidemic and epidemic.vaccination_mode_active then
-        epidemic:markForVaccination(self)
+
+      local function isValidEpidemicTarget(epidemic)
+        return epidemic and epidemic.coverup_selected and
+          (epidemic.vaccination_mode_active or (self.infected and (not self.marked_for_vaccination)))
+      end
+
+      if isValidEpidemicTarget(epidemic) then
```

Comment:
No need to pass in epidemic; you already declared the variable at line 74 above the function =]
(This is why the code test failed)

Reply:
Looks like it was resolved the other way!